src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -668,6 +687,173 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
+		switch healthCertificate.type {
+		case .vaccination:
+			deleteButtonTitle = AppStrings.HealthCertificate.Details.deleteButtonTitle
+		case .test:
+			deleteButtonTitle = AppStrings.HealthCertificate.Details.TestCertificate.primaryButton
+		case .recovery:
+			deleteButtonTitle = AppStrings.HealthCertificate.Details.RecoveryCertificate.primaryButton
+		}
+
+		let footerViewModel = FooterViewModel(
+			primaryButtonName: AppStrings.HealthCertificate.Details.validationButtonTitle,
+			secondaryButtonName: deleteButtonTitle,
+			isPrimaryButtonEnabled: true,
+			isSecondaryButtonEnabled: true,
+			isSecondaryButtonHidden: false,
+			primaryButtonInverted: false,
+			secondaryButtonInverted: true,
+			backgroundColor: .enaColor(for: .cellBackground),
+			secondaryTextColor: .systemRed
+		)
+
+		let footerViewController = FooterViewController(footerViewModel)
+
+		let healthCertificateViewController = HealthCertificateViewController(
+			healthCertifiedPerson: healthCertifiedPerson,
+			healthCertificate: healthCertificate,
+			vaccinationValueSetsProvider: vaccinationValueSetsProvider,
+			dismiss: { [weak self] in
+				self?.dismiss()
+			},
+			didTapValidationButton: { [weak self] in
+				footerViewModel.setLoadingIndicator(true, disable: true, button: .primary)
+				footerViewModel.setLoadingIndicator(false, disable: true, button: .secondary)
+
+				self?.healthCertificateValidationOnboardedCountriesProvider.onboardedCountries { result in
+					footerViewModel.setLoadingIndicator(false, disable: false, button: .primary)
+					footerViewModel.setLoadingIndicator(false, disable: false, button: .secondary)
+
+					switch result {
+					case .success(let countries):
+						self?.showValidationFlow(
+							healthCertificate: healthCertificate,
+							countries: countries
+						)
+					case .failure(let error):
+						self?.showValidationErrorAlert(
+							title: AppStrings.HealthCertificate.Validation.Error.title,
+							error: error
+						)
+					}
+				}
+			},
+			didTapDeleteButton: { [weak self] in
+				self?.showDeleteAlert(
+					certificateType: healthCertificate.type,
+					submitAction: UIAlertAction(
+						title: AppStrings.HealthCertificate.Alert.deleteButton,
+						style: .destructive,
+						handler: { _ in
+							guard let self = self else {
+								Log.error("Could not create strong self")
+								return
+							}
+							self.healthCertificateService.removeHealthCertificate(healthCertificate)
+							let isPersonStillExistent = self.healthCertificateService.healthCertifiedPersons.value.contains(healthCertifiedPerson)
+
+							// Only pop to root if we did not removed the last certificate of a person (because this removes the person, too). A pop would trigger a reload of content which was removed before. If so, dismiss to go back to certificate overview.
+							if isPersonStillExistent {
+								self.navigationController?.popToRootViewController(animated: true)
+							} else {
+								self.dismiss()
+							}
```

Comment:
I think it makes sense to always pop back to the test result on deletion. And probably not back to root, as there could be other view controllers on the stack still.

```suggestion
							self.navigationController?.popViewController(animated: true)
```